qiskit/transpiler/passes/crosstalk_adaptive_schedule.py

```diff
@@ -0,0 +1,667 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Crosstalk mitigation through adaptive instruction scheduling.
+The scheduling algorithm is described in:
+Prakash Murali, David C. Mckay, Margaret Martonosi, Ali Javadi Abhari,
+Software Mitigation of Crosstalk on Noisy Intermediate-Scale Quantum Computers,
+in International Conference on Architectural Support for Programming Languages
+and Operating Systems (ASPLOS), 2020.
+Please cite the paper if you use this pass.
+"""
+
+import math
+import operator
+from itertools import chain, combinations
+from z3 import Real, Bool, Sum, Implies, And, Or, Not, Optimize
```

Comment:
This should be optional, can we change this line to:

```suggestion
try:
    from z3 import Real, Bool, Sum, Implies, And, Or, Not, Optimize
    Z3_AVAIL = True
except ImportError:
    Z3_AVAIL = False
```

Then in the top of the `__init__` method we can check and raise as the first thing like:

```
    def __init__(self, backend_prop, crosstalk_prop, weight_factor=0.5, measured_indices=None):
       super().__init__()
       if not Z3_AVAIL;
           raise ImportError('z3-solver is required to use the crosstalk adaptive schedule')
```

Reply:
Done